lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1239,6 +1242,17 @@ ChangeResult TypeAnalyzer::visitAtenShapeAsTensorOp(
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitNumToTensorOp(PrimNumToTensorScalarOp op) {
+  auto knowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  fillInSizesGivenSizesList(knowledge, {0});
```

Comment:
I don't think this is right. From looking at the definition:

https://github.com/llvm/torch-mlir/blob/8507ce3d4c8f1d7c0042cda5d5d47447b62c67b8/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L618-L621

`fillInSizesGivenSizesList` takes in a `Value` that is the result of a `PrimListConstructOp`.

Additionally, a zero-rank result is represented by the empty list, not a list with one dimension of length zero. Here you can do something similar to what is done for ops that reduce to a scalar:

https://github.com/llvm/torch-mlir/blob/8507ce3d4c8f1d7c0042cda5d5d47447b62c67b8/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L896-L899

Reply:
I must have been typing at the same time as @cathyzhyi. Same comment as @cathyzhyi's